### PR TITLE
Migrate Auth to component framework

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

**Comment by Feiyang1** [comment]
@bojeil-google ping

File: packages/auth/src/exports_auth.js [hunk 1]
```diff
@@ -687,34 +670,43 @@ fireauth.exportlib.exportFunction(
...
-              // the error will be thrown.
-              // One specific use case here is the initialization for the nodejs
-              // client when no API key is provided. This is commonly used
-              // for unauthenticated database access.
-            }
-          }
-        }
-        );
-
+    // Create auth components to register with firebase
```
Comment (bojeil-google) [inline packages/auth/src/exports_auth.js]: nit: Add dot at the end of the comment

File: packages/auth/src/exports_auth.js [hunk 2]
```diff
@@ -687,34 +670,43 @@ fireauth.exportlib.exportFunction(
...
+    const authComponent = { // provides Auth public APIs
```
Comment (bojeil-google) [inline packages/auth/src/exports_auth.js]: Preferably move comment to newline. Capitalize first letter and add dot to end of line.
`// Provides Auth public APIs.`

File: packages/auth/src/exports_auth.js [hunk 3]
```diff
@@ -687,34 +670,43 @@ fireauth.exportlib.exportFunction(
...
+        var app = container['getProvider']('app')['getImmediate']();
+        return new fireauth.Auth(app);
+      },
+      'multipleInstances': false,
+      'serviceProps': namespace,
+      'instantiationMode': 'LAZY',
+      'type':  'PUBLIC'
+    };
+  
+    const authInteropComponent = { // provides Auth internal APIs
```
Comment (bojeil-google) [inline packages/auth/src/exports_auth.js]: Same here

Comment (Feiyang1) [reply packages/auth/src/exports_auth.js]: Done.

File: packages/auth/src/exports_auth.js (hunk 2 above)
Comment (Feiyang1) [reply packages/auth/src/exports_auth.js]: Done.

File: packages/auth/src/exports_auth.js (hunk 1 above)
Comment (Feiyang1) [reply packages/auth/src/exports_auth.js]: Done.

**Pull request merged**